### PR TITLE
User login start

### PR DESCRIPTION
**Pull request by sdkibb**
This will require an npm update.

### CODE REVIEW

File: src/providers/user/user.ts [hunk 1]
```diff
@@ -5,22 +5,70 @@ import { LangCodes, LogProvider } from "../../providers";
...
+  private UserDetails: any; //### When ready, we need to define the user object here
+  private SessionTimeOutMinutes: number = 1;
   private Lang: LangCodes;
   private LangReady: boolean = false;
   private readonly logger: Logger;
 
+  public firstName: string = "";
+  public lastName: string = "";
+  public sessionLastHit: Date;
+  public SIN: string = "";
```
Comment (CalvinRodo) [inline src/providers/user/user.ts]: properties should be camel case to be consistent with others so the first letter should be lower case. 

VSCode and other tools don't deal too well with upper case acronyms it thinks that the capital letters demark the beginning of  a word so all subsequent letters after the first in an acronym should be lower case so this should be 

sin not SIN

File: src/providers/user/user.ts [hunk 2]
```diff
@@ -5,22 +5,70 @@ import { LangCodes, LogProvider } from "../../providers";
...
+  public passCode: string = "";
+  public region = "";
+  public language = ""; //### The chosen user language on the API account may differ from the user chosen mobile app language, that's why we have two language properties
+  public sessionID = "";
```
Comment (CalvinRodo) [inline src/providers/user/user.ts]: Acronyms should be camel case

File: src/providers/user/user.ts [hunk 3]
```diff
@@ -120,41 +168,109 @@ export class User {
...
+      accesscode: _AccessCode,
+      region: _Region,
+      language: _Language
+    };
+
+    this.api.post("authentication", accountInfo).subscribe(
+      (Response: any) => {
+        //### TODO - Check for non-existants of expected properties (e.g. AuthResponseStatus
+        if (Response.AuthResponseStatus === "Success") {
+          let _UserDetails = {
```
Comment (CalvinRodo) [inline src/providers/user/user.ts]: might be worth breaking this object assignment out to a method.

File: src/providers/user/user.ts [hunk 4]
```diff
@@ -120,41 +168,109 @@ export class User {
...
+    );
   }
 
   logout() {
-    this._user = null;
+    this.UserDetails = null;
+  }
+
+  private _loggedIn(resp: any, SaveLocalStorage: boolean = false) {
+    this.firstName = resp.firstName;
```
Comment (CalvinRodo) [inline src/providers/user/user.ts]: Might be worth breaking out these assignments to a method.

**Comment by sdkibb** [comment]
I'll do the changes first.

**Comment by sdkibb** [comment]
Updated based on comments. Please re-review.

**Pull request merged**